src/GovUk.Education.ExploreEducationStatistics.Data.Api/Models/Query/ObservationPredicateBuilder.cs

```diff
@@ -0,0 +1,239 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Linq.Expressions;
+using GovUk.Education.ExploreEducationStatistics.Data.Api.Extensions;
+using GovUk.Education.ExploreEducationStatistics.Data.Api.Services;
+using GovUk.Education.ExploreEducationStatistics.Data.Model;
+using GovUk.Education.ExploreEducationStatistics.Data.Model.Database;
+using GovUk.Education.ExploreEducationStatistics.Data.Model.Query;
+
+namespace GovUk.Education.ExploreEducationStatistics.Data.Api.Models.Query
+{
+    public static class ObservationPredicateBuilder
+    {
+        public static Expression<Func<Observation, bool>> Build(SubjectMetaQueryContext query)
+        {
+            var predicate = PredicateBuilder.True<Observation>()
+                .And(observation => observation.SubjectId == query.SubjectId);
+
+            if (query.TimePeriod != null)
+            {
+                // Don't use the observation.GetTimePeriod() extension in the expression here as it can't be translated
+                predicate = predicate.And(observation => GetTimePeriodRange(query.TimePeriod)
+                    .Contains(observation.Year + "_" + observation.TimeIdentifier));
+            }
+
+            if (query.GeographicLevel != null)
+            {
+                predicate = predicate.And(observation => observation.GeographicLevel == query.GeographicLevel);
+            }
+
+            if (ObservationalUnitExists(query))
+            {
+                predicate = predicate.And(ObservationalUnitsPredicate(query));
+            }
+
+            return predicate;
+        }
+
+        private static Expression<Func<Observation, bool>> ObservationalUnitsPredicate(SubjectMetaQueryContext query)
+        {
+            var predicate = PredicateBuilder.False<Observation>();
+
+            if (query.Country != null)
```

Comment:
could use ? / ?? here maybe although I think there's also an object.IsNull() - I guess all the rearenders should probs be consistent about which we use when but we can agree that at the next retro maybe.

Reply:
I need to run the importer to test this change with data. I don't currently have any data so I'm going to make this change later on.